File: src/main/java/com/paulrezzonico/util/QuoteManager.java
package com.paulrezzonico.util;

import com.paulrezzonico.DataProvider.IDataProvider;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

import java.util.List;
import java.util.Random;

@Component
public class QuoteManager {

    @Autowired
    private IDataProvider dataProvider;

    public String getRandomQuote() {

        List<String> quotes = dataProvider.getData("/voiceLines.txt");

        if (!quotes.isEmpty()) {
            Random random = new Random();
            return quotes.get(random.nextInt(quotes.size()));
        }

        return "No quotes found";
    }
}

File: src/main/java/com/paulrezzonico/util/QuoteManager.kt
package com.paulrezzonico.util

import com.paulrezzonico.dataProvider.IDataProvider
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Component
import java.util.*

@Component
class QuoteManager {
    @Autowired
    private val dataProvider: IDataProvider? = null

    val randomQuote: String
        get() {
            val quotes = dataProvider!!.getData("/voiceLines.txt")

            if (quotes.isNotEmpty()) {
                val random = Random()
                return quotes[random.nextInt(quotes.size)]
            }

            return "No quotes found"
        }
}
